snuba/datasets/plans/sliced_storage.py

```diff
@@ -49,11 +52,47 @@ def select_cluster(
         raise NotImplementedError
 
 
+def _should_use_mega_cluster(
+    storage_set: StorageSetKey, logical_partition: int
+) -> bool:
+    """
+    Helper method to find out whether a logical partition of a sliced storage
+    set needs to send queries to the mega cluster.
+
+    Mega cluster needs to be used when a logical partition's data might be
+    across multiple slices. This happens when a logical partition is mapped
+    to a new slice. In such cases, the old data resides in some different
+    slice than what the new mapping says.
+    """
+    key = f"{MEGA_CLUSTER_RUNTIME_CONFIG_PREFIX}_{storage_set.value}"
```

Comment:
nit: runtime config keys are lower case

Reply:
The actual runtime prefix is lower case. The constant is defined in upper case.